cmd/tink-worker/internal/registry_test.go

```diff
@@ -0,0 +1,72 @@
+package internal
+
+import (
+	"context"
+	"errors"
+	"io"
+	"io/ioutil"
+	"strings"
+	"testing"
+
+	"github.com/docker/docker/api/types"
+	"github.com/packethost/pkg/log"
+	"github.com/stretchr/testify/assert"
+)
+
+func setupTestLogger() log.Logger {
+	service := "github.com/tinkerbell/tink"
+	logger, err := log.Init(service)
+	if err != nil {
+		panic(err)
+	}
+	return logger
+}
+
+type imagePullerMock struct{}
+
+func (d *imagePullerMock) ImagePull(ctx context.Context, str string, op types.ImagePullOptions) (io.ReadCloser, error) {
+	var s string
+	var err error
+	if str == "test/success" {
+		s = "{\"status\": \"hello\",\"error\":\"\"}{\"status\":\"world\",\"error\":\"\"}"
```

Comment:
There should only be a test case when decoding fails, as IMO it's important part of the spec, to decide on what to do if Docker throws some unexpected message.

Reply:
second test case does that. `Tested, failure of the image pull`

Reply:
third case does that. In that scenario, while reading the stream docker will bail out with `No space left on device`